lidiabrowser/lidia/models.py

```diff
@@ -0,0 +1,89 @@
+from django.db import models
+
+
+class Publication(models.Model):
+    zotero_id = models.CharField(max_length=100, unique=True)
+    attachment_id = models.CharField(max_length=16, unique=True, null=True)
+    title = models.CharField(max_length=255, null=True)
+
+    def __str__(self):
+        return self.title or self.zotero_id
+
+
+class Language(models.Model):
+    code = models.CharField(max_length=3, unique=True)
+    name = models.CharField(max_length=100, null=True)
+
+    def __str__(self):
+        return self.name or self.code
+
+
+class Annotation(models.Model):
+    RELATION_TYPE_CHOICES = [
+        ("", "None"),
+        ("contradicts", "Contradicts"),
+        ("generalizes", "Generalizes"),
+        ("invalidates", "Invalidates"),
+        ("specialcase", "Is a special case of"),
+        ("supports", "Supports"),
+    ]
+
+    zotero_id = models.CharField(max_length=100, unique=True, null=False)
+    parent_attachment = models.ForeignKey(Publication, on_delete=models.CASCADE, to_field='attachment_id', blank=True, null=True)
+    textselection = models.TextField(default='') # Limit how much text we store, or limit displayed text?
+    argname = models.CharField(max_length=100, unique=True, null=True) # Should we enforce unique here or allow and warn?
```

Comment:
I did not know that `unique=True` and `null=True` were allowed together! I would rather not enforce it and indeed warn, because we cannot enforce it in the extension. I think it is better if 'incorrect' annotations appear in Django and that users can see that something is wrong with them.

If we don't enforce uniqueness, I would also make this field non-nullable and store empty strings instead, as Django recommends.

Reply:
Yeah, if the unique values are not guaranteed to be present it's even a necessity. But agree, will change it.